meta-resin-common/recipes-bsp/u-boot/patches/resin-specific-env-configuration.patch

```diff
@@ -105,13 +187,20 @@ index 0000000..2ff0fa5
 +               "fi;\0" \
 +       \
 +       "resin_set_kernel_root=" \
-+               "run resin_set_dev_index;" \
++               "if test -n \"${resin_boot_from_root}\"; then " \
++                       "setenv resin_loader ext4load;" \
++                       "setenv resin_scan_parts ${resin_root_parts};" \
```

Comment:
It seems like something is wrong here. This will make resin_scan_devs_and_parts to scan for the flasher file flag in the root partitions. The image type file flags are only available in the boot partitions.

Reply:
As well, above, `resin_flasher_detect` uses `root_part_index` which is again wrong. Could you explain what is this supposed to do?